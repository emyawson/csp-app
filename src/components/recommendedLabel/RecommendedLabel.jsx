import React from 'react';
import './RecommendedLabel.scss';
import { phrase } from '../../utils/phrase';

const RecommendedLabel = () => (
        <p className="label">{phrase.recommendedLabel}</p>
);

export default RecommendedLabel;
