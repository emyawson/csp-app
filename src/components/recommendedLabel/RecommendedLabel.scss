@import '../../index';

.label {
  color: $colour-semantic-positive;
  font-size: $font-S;
  line-height: 159.5%;
  margin-bottom: 8px;
};
