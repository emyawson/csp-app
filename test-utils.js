import React from 'react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import { Router } from 'react-router-dom';
import { connectRouter } from 'connected-react-router';
import { render } from '@testing-library/react';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import { history } from './src/store';
import reducers from './src/store/reducer/index';

const createRootReducer = (reducerHistory) => combineReducers({
  router: connectRouter(reducerHistory),
  ...reducers,
});

// Setup redux persist
const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
};
const persistedReducer = persistReducer(
  persistConfig,
  createRootReducer(history),
);

const renderWithRouter = (ui, options = {}, initialState) => ({
  ...render(
      <Provider store={createStore(persistedReducer, initialState)}>
        <Router history={history}>{ui}</Router>
      </Provider>,
  ),
});

const AllTheProviders = () => {};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, renderWithRouter };
