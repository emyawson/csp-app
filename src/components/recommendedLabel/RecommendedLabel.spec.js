import React from 'react';
// import { render, screen } from "@testing-library/react";
import RecommendedLabel from './RecommendedLabel.jsx';
// import { renderWithRouter } from "../test-utils";
import { renderWithRouter } from '../../../test-utils';

const defaultState = {};

const renderComponent = (state = {}) => {
  const combinedState = {
    ...defaultState,
    ...state,
  };
  return renderWithRouter(<RecommendedLabel />, combinedState);
};

test('renders <RecommendedLabel/>', () => {
  const { container } = renderComponent();
  expect(container).toMatchSnapshot();

  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
});
