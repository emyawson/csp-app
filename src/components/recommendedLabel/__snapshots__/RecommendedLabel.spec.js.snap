// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <RecommendedLabel/> 1`] = `
<div>
  <p
    class="label"
  >
    Aanbevolen
  </p>
</div>
`;
